src/adaptors/tcp_adaptor.c

```diff
@@ -116,6 +120,26 @@ static inline uint64_t qdr_tcp_conn_linkid(const qdr_tcp_connection_t *conn)
     return conn->instream ? conn->incoming_id : conn->outgoing_id;
 }
 
+static inline const char * qdr_link_direction_name(const qdr_link_t *link)
+{
+    assert(link);
+    return qdr_link_direction(link) == QD_OUTGOING ? "outgoing" : "incoming";
+}
+
+static inline const char * qdr_tcp_connection_role_name(const qdr_tcp_connection_t *tc)
+{
+    assert(tc);
+    return tc->ingress ? "client" : "server";
```

Comment:
I find the use of client and server confusing here. What you call the client is actually the server side in tcp terms. I think ingress/egress is less ambiguous (perhaps qualified by bridge, e.g. 'ingress bridge') and is already used elsewhere even with this patch.

Reply:
Ambiguity abounds no matter how it is described. I figure that a client originates a connection to a listener and that a connector originates a connection to a server. Clients never accept connections and servers never originate connections. 
Also, there was some bad code logic in these descriptions that caused the logs to say wrong things. An update is coming shortly.

Reply:
> Clients never accept connections and servers never originate connections.

But that is not what the line above implies, is it? If ingress=true it is a connection managed by a TcpListener, i,e, it was an accepted connection.